doc/pserver.md

```diff
@@ -0,0 +1,226 @@
+# Parameter Server Design
+
+
+## Overview
+
+
+## PServer
+
+
+### Compoments
+
+
+![pserver](./images/pserver.png)
+
+
+PServer contains three main compoments:
+
+- KVStore
+- Gradient Queue
+- Optimizer
+
+1. The worker initializes a model, and pushes parameters to KVStore.
+2. Before each step of training, the worker pulls the latest model from KVStore.
+3. After forward/backward computation, the worker push gradients to Gradient Queue waiting for processing.
```

Comment:
Please add a proper `a` or `the` at multiple places here.

Reply:
Done